### PR TITLE
OpenID

### PR DESCRIPTION
**Pull request by idiidk**
Switch to my [magister-openid](https://www.npmjs.com/package/magister-openid) package to support proper OpenID authentication and token management.

Where you could locally store the token variable, (that would expire after an hour xd), you should now save the tokenSet. This is an object containing the access and refresh tokens. You can initialize MagisterJS with this token set, tokens are automatically checked and refreshed if necessary. This negates the saving of plaintext passwords!

My package is a wrapper around the [openid-client](https://www.npmjs.com/package/openid-client) library. It just adds support for Magister.
I think it's a lot more maintainable and clean to keep the authentication code and main library code separate. So that's why I created the package. 

All tests pass on my end, and the code is linted to spec.

CREDIT TO [@Sj3rd](https://github.com/sj3rd) FOR HELPING ME FIGURE THIS OUT, COULDN'T HAVE DONE IT WITHOUT HIM (en z'n papa is pliesie dus deze credit moet wel)

x idiidk

### CODE REVIEW

**Comment by idiidk** [comment]
Ly @netlob

**Comment by Sj3rd** [comment]
Ly2bb @idiika

**Comment by idiidk** [comment]
This will also fix #100 and #91

**Comment by lieuwex** [comment]
Super nice!

I totally ship you guys too.

File: test/test.js [hunk 1]
```diff
@@ -100,13 +100,13 @@ describe('Magister', function() {
 			...options,
 			username: 'xxx',
 			password: 'xxx',
-		})).to.be.rejectedWith(magisterjs.AuthError)
```
Comment (lieuwex) [inline test/test.js]: Is it still possible to distinguish authentication errors from general errors?

Comment (idiidk) [reply test/test.js]: My library returns a generic error, so it's not possible in the current state of the commit. I can wrap the error in an autherror if that's necessary

File: src/http.js [hunk 2]
```diff
@@ -72,13 +71,15 @@ class Http {
 	 * @param {Object} obj
 	 * @returns {Request}
 	 */
-	makeRequest(obj) {
+	async makeRequest(obj) {
```
Comment (jvdoorn) [inline src/http.js]: Does this (the function now being async) create any issues?

File: src/http.js [hunk 3]
```diff
@@ -72,13 +71,15 @@ class Http {
...
+		await this.authManager.checkExpiration()
+		const accessToken = this.authManager.accessToken
 		const init = {
 			method: obj.method,
 			timeout: this._requestTimeout,
 			headers: {
 				...obj.headers,
-				Authorization: 'Bearer ' + this._token,
+				Authorization: `Bearer ${accessToken}`,
```
Comment (jvdoorn) [inline src/http.js]: Much cleaner.

File: src/http.js (hunk 2 above)
Comment (idiidk) [reply src/http.js]: Nope, I fixed all the refrences

File: test/test.js (hunk 1 above)
Comment (lieuwex) [reply test/test.js]: It's fine for now, but maybe nice in the future. We're lucky the library is still in alpha state so the API isn't fixed. But I think it's nice to add the AuthError before releasing v2 as stable.

**Comment by idiidk** [comment]
Okay, the login function now throws AuthErrors and the tests are updated to support that!

**Comment by LevitatingBusinessMan** [comment]
I love this!

**Comment by Sj3rd** [comment]
Great to hear all this work wasn't for nothing! :)

**Comment by lieuwex** [comment]
Rebased on top of master with some squashes!
Thank you very much :)

**Pull request closed without merging**